# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/ipc/GamepadTestChannelParent.cpp
# Commit: 349854c5ad56
# Full Hash: 349854c5ad56f0885a5b84ba80f881b9ac9900ff
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-12-03 09:47:26
# Regressor Bug: 1657404
# File Overlap Count: 1
# Description:
#   Bug 1657404 - Gamepad: Make a singleton to hold monitoring observers r=handyman
#   
#   Currently, the list of monitoring observers is stored in the
#   GamepadPlatformService. But it's possible for testing to start before an
#   event channel has been created, or to exist longer. That could result in the
# ==============================================================================

diff -r 117da9de7f66 -r 349854c5ad56 dom/gamepad/ipc/GamepadTestChannelParent.cpp
--- a/dom/gamepad/ipc/GamepadTestChannelParent.cpp	Wed Dec 02 23:18:38 2020 +0000
+++ b/dom/gamepad/ipc/GamepadTestChannelParent.cpp	Wed Dec 02 22:32:30 2020 +0000
@@ -19,20 +19,12 @@
 
 GamepadTestChannelParent::GamepadTestChannelParent() {
   ::mozilla::ipc::AssertIsOnBackgroundThread();
-  RefPtr<GamepadPlatformService> service =
-      GamepadPlatformService::GetParentService();
-  MOZ_ASSERT(service);
-
-  service->GetMonitoringState().AddObserver(this);
+  GamepadMonitoringState::GetSingleton().AddObserver(this);
 }
 
 GamepadTestChannelParent::~GamepadTestChannelParent() {
   ::mozilla::ipc::AssertIsOnBackgroundThread();
-  RefPtr<GamepadPlatformService> service =
-      GamepadPlatformService::GetParentService();
-  MOZ_ASSERT(service);
-
-  service->GetMonitoringState().RemoveObserver(this);
+  GamepadMonitoringState::GetSingleton().RemoveObserver(this);
 }
 
 void GamepadTestChannelParent::AddGamepadToPlatformService(
@@ -80,7 +72,7 @@
   // started. Everything else won't be deferred and will fail if monitoring
   // isn't running
   if (body.type() == GamepadChangeEventBody::TGamepadAdded) {
-    if (service->GetMonitoringState().IsMonitoring()) {
+    if (GamepadMonitoringState::GetSingleton().IsMonitoring()) {
       AddGamepadToPlatformService(aID, body.get_GamepadAdded());
     } else {
       mDeferredGamepadAdded.AppendElement(
@@ -89,7 +81,7 @@
     return IPC_OK();
   }
 
-  if (!service->GetMonitoringState().IsMonitoring()) {
+  if (!GamepadMonitoringState::GetSingleton().IsMonitoring()) {
     return IPC_FAIL(this, "Simulated message received while not monitoring");
   }
 
